src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -118,7 +118,8 @@ def pull_helm_chart(registry_path, chart_export_path, kube_config, kube_context,
     chart_version = registry_path.split(':')[1]
 
     if new_path:
-        if (version.parse(chart_version) < version.parse("1.14.0")):
+        # Version check for stable release train (chart_version will be in X.Y.Z format as opposed to X.Y.Z-NONSTABLE)
+        if '-' not in chart_version and (version.parse(chart_version) < version.parse("1.14.0")):
```

Comment:
Please note that the CLI team has defined a unified new extension version upgrade rule, please take a look https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md#cli-extension-version-transition-table